packages/zip/src/buffer-reader.js

```diff
@@ -3,8 +3,21 @@
 
 const q = JSON.stringify;
 
+/**
+ * @typedef {object} BufferReaderState
+ * @property {Uint8Array} bytes
+ * @property {DataView} data
+ * @property {number} length
+ * @property {number} index
+ * @property {number} offset
+ */
+
+/** @type {WeakMap<BufferReader, BufferReaderState>} */
 const privateFields = new WeakMap();
 
+/** @type {(bufferReader: BufferReader) => BufferReaderState} */
+const privateFieldsGet = privateFields.get.bind(privateFields);
```

Comment:
why this change?

Reply:
To get the typing right in one location (`privateFields.get(this).length` etc. are rejected by TypeScript because it doesn't know that `this` is guaranteed to be a key of `privateFields`).

Reply:
Ok. Probably better than trying to make it work with `ThisType`. But consider a source code comment with what you explained here